ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp

```diff
@@ -836,15 +836,19 @@ class TGraceJoinSpillingSupportState : public TComputationValue<TGraceJoinSpilli
 void DoCalculateWithSpilling(TComputationContext& ctx) {
     UpdateSpilling();
 
-    if (!HasMemoryForProcessing() && !IsSpillingFinalized) {
-        bool isWaitingForReduce = TryToReduceMemoryAndWait();
-        if (isWaitingForReduce) return;
-    }
-
+    ui64 cntrows = 0;
     while (*HaveMoreLeftRows || *HaveMoreRightRows) {
+        if (0 == (cntrows++ % CheckMemoryRowLimit)) { // checks before first loop and
```

Comment:
`cntrows` redundant. We can easily check memory on each row this call is as simple as checking the value of bool variable.

Reply:
Ok